### ISSUE TITLE
Plugin fails to load

### ISSUE THREAD

**Issue by SiliconExarch** [reckhou/MidiBard#32]
**Describe the bug**
The plugin encounters an error while loading on Dalamud 7.4.2.0

**To Reproduce**
1. Install/Enable the plugin
2. "This plugin failed to load. Please contact the author for more information."

**Expected behavior**
The plugin loads successfully

**Screenshots**
![image](https://user-images.githubusercontent.com/25375403/213408666-01d618f1-ba88-43f6-9eea-45893cf71d81.png)

**MidiBard Version:**
 
 - 2.3.1.3

**Additional context**
```
2023-01-19 08:04:31.062 +00:00 [INF] [MidiBard2] [OffsetManager][DoPerformAction] [ffxiv_dx11.exe+0x1338080]
2023-01-19 08:04:31.103 +00:00 [INF] [MidiBard2] [OffsetManager][InstrumentOffset] System.Byte 20
2023-01-19 08:04:31.106 +00:00 [INF] [MidiBard2] [OffsetManager][ApplyGraphicConfigsFunc] 7FF6433A0AF0
2023-01-19 08:04:31.195 +00:00 [INF] [MidiBard2] [OffsetManager][ApplyGraphicConfigsFunc] [ffxiv_dx11.exe+0x260AF0]
2023-01-19 08:04:31.210 +00:00 [INF] [MidiBard2] [OffsetManager][NetworkEnsembleStart] 7FF643BD7990
2023-01-19 08:04:31.307 +00:00 [INF] [MidiBard2] [OffsetManager][NetworkEnsembleStart] [ffxiv_dx11.exe+0xA97990]
2023-01-19 08:04:31.311 +00:00 [INF] LASTEXCEPTION:eyJXaGVuIjoiMjAyMy0wMS0xOVQwODowNDozMS4zMTExMjY4KzAwOjAwIiwiSW5mbyI6IlN5c3RlbS5SZWZsZWN0aW9uLlRhcmdldEludm9jYXRpb25FeGNlcHRpb246IEV4Y2VwdGlvbiBoYXMgYmVlbiB0aHJvd24gYnkgdGhlIHRhcmdldCBvZiBhbiBpbnZvY2F0aW9uLlxyXG4gLS0tPiBTeXN0ZW0uVHlwZUluaXRpYWxpemF0aW9uRXhjZXB0aW9uOiBUaGUgdHlwZSBpbml0aWFsaXplciBmb3IgJ01pZGlCYXJkLkFnZW50TWFuYWdlcicgdGhyZXcgYW4gZXhjZXB0aW9uLlxyXG4gLS0tPiBTeXN0ZW0uTWlzc2luZ0ZpZWxkRXhjZXB0aW9uOiBGaWVsZCBub3QgZm91bmQ6ICdGRlhJVkNsaWVudFN0cnVjdHMuRkZYSVYuQ2xpZW50LlVJLkFnZW50LkFnZW50TW9kdWxlLkFnZW50QXJyYXknLlxyXG4gICBhdCBNaWRpQmFyZC5BZ2VudE1hbmFnZXIuLmN0b3IoKVxyXG4gICBhdCBNaWRpQmFyZC5BZ2VudE1hbmFnZXIuLmNjdG9yKCkgaW4gRDpcXEdpdEh1YlxcTWlkaUJhcmRcXE1pZGliYXJkXFxNYW5hZ2Vyc1xcQWdlbnRzXFxBZ2VudE1hbmFnZXIuY3M6bGluZSA0N1xyXG4gICAtLS0gRW5kIG9mIGlubmVyIGV4Y2VwdGlvbiBzdGFjayB0cmFjZSAtLS1cclxuICAgYXQgTWlkaUJhcmQuTWlkaUJhcmQuLmN0b3IoRGFsYW11ZFBsdWdpbkludGVyZmFjZSBwaSwgQ2hhdEd1aSBjaGF0R3VpKSBpbiBEOlxcR2l0SHViXFxNaWRpQmFyZFxcTWlkaWJhcmRcXE1pZGlCYXJkLmNzOmxpbmUgMTE1XHJcbiAgIGF0IFN5c3RlbS5SdW50aW1lTWV0aG9kSGFuZGxlLkludm9rZU1ldGhvZChPYmplY3QgdGFyZ2V0LCBWb2lkKiogYXJndW1lbnRzLCBTaWduYXR1cmUgc2lnLCBCb29sZWFuIGlzQ29uc3RydWN0b3IpXHJcbiAgIGF0IFN5c3RlbS5SZWZsZWN0aW9uLkNvbnN0cnVjdG9ySW52b2tlci5JbnZva2UoT2JqZWN0IG9iaiwgSW50UHRyKiBhcmdzLCBCaW5kaW5nRmxhZ3MgaW52b2tlQXR0cilcclxuICAgLS0tIEVuZCBvZiBpbm5lciBleGNlcHRpb24gc3RhY2sgdHJhY2UgLS0tXHJcbiAgIGF0IFN5c3RlbS5SZWZsZWN0aW9uLkNvbnN0cnVjdG9ySW52b2tlci5JbnZva2UoT2JqZWN0IG9iaiwgSW50UHRyKiBhcmdzLCBCaW5kaW5nRmxhZ3MgaW52b2tlQXR0cilcclxuICAgYXQgU3lzdGVtLlJlZmxlY3Rpb24uUnVudGltZUNvbnN0cnVjdG9ySW5mby5JbnZva2UoT2JqZWN0IG9iaiwgQmluZGluZ0ZsYWdzIGludm9rZUF0dHIsIEJpbmRlciBiaW5kZXIsIE9iamVjdFtdIHBhcmFtZXRlcnMsIEN1bHR1cmVJbmZvIGN1bHR1cmUpXHJcbiAgIGF0IFN5c3RlbS5SZWZsZWN0aW9uLk1ldGhvZEJhc2UuSW52b2tlKE9iamVjdCBvYmosIE9iamVjdFtdIHBhcmFtZXRlcnMpXHJcbiAgIGF0IERhbGFtdWQuSW9DLkludGVybmFsLlNlcnZpY2VDb250YWluZXIuQ3JlYXRlQXN5bmMoVHlwZSBvYmplY3RUeXBlLCBPYmplY3RbXSBzY29wZWRPYmplY3RzKSBpbiBDOlxcZ29hdHNvZnRcXGNvbXBhbnlzZWNyZXRzXFxkYWxhbXVkXFxJb0NcXEludGVybmFsXFxTZXJ2aWNlQ29udGFpbmVyLmNzOmxpbmUgMTA0XHJcbiAgIGF0IERhbGFtdWQuUGx1Z2luLkludGVybmFsLlR5cGVzLkxvY2FsUGx1Z2luLkxvYWRBc3luYyhQbHVnaW5Mb2FkUmVhc29uIHJlYXNvbiwgQm9vbGVhbiByZWxvYWRpbmcpIGluIEM6XFxnb2F0c29mdFxcY29tcGFueXNlY3JldHNcXGRhbGFtdWRcXFBsdWdpblxcSW50ZXJuYWxcXFR5cGVzXFxMb2NhbFBsdWdpbi5jczpsaW5lIDQyMiIsIkNvbnRleHQiOiJbTE9DQUxQTFVHSU5dIEVycm9yIHdoaWxlIGxvYWRpbmcgTWlkaUJhcmQgMiJ9
2023-01-19 08:04:31.310 +00:00 [ERR] [LOCALPLUGIN] Error while loading MidiBard 2
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.TypeInitializationException: The type initializer for 'MidiBard.AgentManager' threw an exception.
 ---> System.MissingFieldException: Field not found: 'FFXIVClientStructs.FFXIV.Client.UI.Agent.AgentModule.AgentArray'.
   at MidiBard.AgentManager..ctor()
   at MidiBard.AgentManager..cctor() in D:\GitHub\MidiBard\Midibard\Managers\Agents\AgentManager.cs:line 47
   --- End of inner exception stack trace ---
   at MidiBard.MidiBard..ctor(DalamudPluginInterface pi, ChatGui chatGui) in D:\GitHub\MidiBard\Midibard\MidiBard.cs:line 115
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 104
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 422
   at Dalamud.Plugin.Internal.PluginManager.LoadPluginAsync(FileInfo dllFile, LocalPluginManifest manifest, PluginLoadReason reason, Boolean isDev, Boolean isBoot, Boolean doNotLoad) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\PluginManager.cs:line 871
2023-01-19 08:04:31.311 +00:00 [INF] [PLUGINM] Regular plugin failed to load, adding anyways: MidiBard2.dll
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.TypeInitializationException: The type initializer for 'MidiBard.AgentManager' threw an exception.
 ---> System.MissingFieldException: Field not found: 'FFXIVClientStructs.FFXIV.Client.UI.Agent.AgentModule.AgentArray'.
   at MidiBard.AgentManager..ctor()
   at MidiBard.AgentManager..cctor() in D:\GitHub\MidiBard\Midibard\Managers\Agents\AgentManager.cs:line 47
   --- End of inner exception stack trace ---
   at MidiBard.MidiBard..ctor(DalamudPluginInterface pi, ChatGui chatGui) in D:\GitHub\MidiBard\Midibard\MidiBard.cs:line 115
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   --- End of inner exception stack trace ---
   at System.Reflection.ConstructorInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
   at System.Reflection.RuntimeConstructorInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Dalamud.IoC.Internal.ServiceContainer.CreateAsync(Type objectType, Object[] scopedObjects) in C:\goatsoft\companysecrets\dalamud\IoC\Internal\ServiceContainer.cs:line 104
   at Dalamud.Plugin.Internal.Types.LocalPlugin.LoadAsync(PluginLoadReason reason, Boolean reloading) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\Types\LocalPlugin.cs:line 422
   at Dalamud.Plugin.Internal.PluginManager.LoadPluginAsync(FileInfo dllFile, LocalPluginManifest manifest, PluginLoadReason reason, Boolean isDev, Boolean isBoot, Boolean doNotLoad) in C:\goatsoft\companysecrets\dalamud\Plugin\Internal\PluginManager.cs:line 871
```

**Comment by akira0245** [comment]
/xldev -> Open dalamud branch switcher -> select `release` branch -> Pick & Restart. The plugin will update when dalamud release(not stg) branch applied there changes

**Comment by SiliconExarch** [comment]
Fixed as of version 2.4.0.1.
